llvm-external-projects/iree-dialects/test/Dialect/iree_vector_ext/roundtrip.mlir

```diff
@@ -0,0 +1,19 @@
+// RUN: iree-dialects-opt --split-input-file %s | FileCheck %s
+
+#row_layout1 = #iree_vector_ext.per_dim_layout<"BatchX"<"LaneX"<"VecY", 2>, 4>, 4>
+#col_layout1 = #iree_vector_ext.per_dim_layout<"BatchY"<"LaneY"<"VecX", 4>, 2>, 4>
+#layout1 = #iree_vector_ext.layout<#row_layout1, #col_layout1>
```

Comment:
Could you please help understand what this means? Thanks in advance :)

Reply:
Hi Diego! Thanks for taking a look. Yes, let me try and explain -
Say we have a SIMD vector of shape <32x32xf16> and we would like to lower this using the
amdgpu.mfma 32x32x8 operation. The amdgpu.mfma operation specifies that each thread should hold 4 entries of the 32x8 a matrix (as there are 64 threads). So we use the above information to encode how to map the entries of the A matrix to the entries to be loaded by a given thread.
This is shown below.

<img width="673" alt="Screenshot 2023-11-15 at 10 09 58 AM" src="https://github.com/openxla/iree/assets/3144552/08bda633-4562-47c5-ae4f-e157e5934eab">
 
Now if you look along a single row, you can see that we can think of creating two new dimensions (say VectorX = 4 and LaneX = 2) so that we can represent the mapping from lane_y to col_offset as

```
col_offset = 4 * lane_x + vector_x
```

What this means is that for a given threadId.x (which is lane_x) we can determine what indices to load by using the above formula and iterating over (vector_x = 0, 1, 2, 3). We can repeat this process for the other tensor dimension as well. 

Hope that helps!